lib/handler/mruby.c

```diff
@@ -405,13 +453,14 @@ mrb_value prepare_fibers(h2o_mruby_context_t *ctx)
     return result;
 }
 
-static void on_context_init(h2o_handler_t *_handler, h2o_context_t *ctx)
+static void on_context_init(h2o_handler_t *_handler, h2o_context_t *ctx, h2o_pathconf_t *pathconf)
 {
     h2o_mruby_handler_t *handler = (void *)_handler;
     h2o_mruby_context_t *handler_ctx = h2o_mem_alloc(sizeof(*handler_ctx));
 
     handler_ctx->handler = handler;
     handler_ctx->shared = get_shared_context(ctx);
+    handler_ctx->pathconf = pathconf;
```

Comment:
Why does `pathconf` need to be a per loop variable? It is not thread-dependent, and I believe that you can make it a property of `h2o_mruby_handler_t`. The added bonus of making such change will be that you would be able to record the address in `h2o_mruby_register` and you no longer need to change the type of `on_context_init`.

Reply:
@kazuho Great idea, thx. I fixed it at https://github.com/h2o/h2o/pull/1217/commits/bc68c17b98b70b27150dc9a46f126b28ef47a797